felix/cachingmap/caching_map.go

```diff
@@ -260,16 +264,16 @@ func (c *CachingMap[K, V]) ApplyUpdatesOnly() error {
 // ApplyDeletionsOnly applies any pending deletions to the dataplane map.  It doesn't add or update any keys that
 // are new/changed.
 func (c *CachingMap[K, V]) ApplyDeletionsOnly() error {
-	logrus.WithField("name", c.name).Debug("Applying deletions to BPF map.")
+	logrus.WithField("name", c.name).Debug("Applying deletions to DP map.")
 	err := c.maybeLoadCache()
 	if err != nil {
 		return err
 	}
 	var errs ErrSlice
 	for k := range c.pendingDeletions {
 		err := c.dpMap.Delete(k)
-		if err != nil && !bpf.IsNotExists(err) {
-			logrus.WithError(err).Warn("Error while deleting from BPF map")
+		if err != nil && err != ErrNotExists {
```

Comment:
Annoying little problem dealing with different exception types that mean the same thing.  Wonder if we should pass in an `shouldIgnoreDeletionError(error) bool` func or maybe change the code we use it with to not return a not exists error?

Reply:
This gave me a hard time, and I originally though (and forgot to do) that the bpf.TypedMap would return this error on behalf of the bpf.Map. And that any other user would return this error as well. But I like the idea that the cachingmap.DataplaneMap would have `ShouldIgnoreDeletionError` or `IsEExist()` or something like that.